### ISSUE TITLE
Type of `type` incorrectly narrowed to `type[object]`

### ISSUE THREAD

**Issue by matejcik** [microsoft/pyright#9656]
If I call `issubclass` on an unbound `type` variable (or `type[Any]`), then the else-branch will narrow it to `type[object]`.
This will then cause an error when using `my_type(somevalue)` to instantiate that type.

```python
def get_type() -> type:
    return int

my_type = get_type()

reveal_type(my_type)  # -> type
if issubclass(my_type, str):
    print("weird")
else:
    reveal_type(my_type)  # -> type[object]
    foobar = my_type(42)
```
output:
```
/home/matejcik/test.py
  /home/matejcik/test.py:6:13 - information: Type of "my_type" is "type"
  /home/matejcik/test.py:10:17 - information: Type of "my_type" is "type[object]"
  /home/matejcik/test.py:11:22 - error: Expected 0 positional arguments (reportCallIssue)
1 error, 0 warnings, 2 informations 

```

**VS Code extension or command-line**
VScode extension in current version and pyright **1.1.391**

**Comment by erictraut** [comment]
I think that narrowing to `type[object]` is technically correct in this case, but I agree that it would be better to evaluate to `type[Unknown]` (or just `type` which is equivalent).

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.392.
